### ISSUE TITLE
Nested selector doesn't seem to work

### ISSUE THREAD

**Issue by NathanC** [jbetancur/react-data-table-component#1022]
## Issue Check list
- [x] Agree to the [Code of Conduct](https://github.com/jbetancur/react-data-table-component/blob/master/CODE-OF-CONDUCT.md)
- [x] Read the README
- [x] You are using React 16.8.0+
- [x] You installed `styled-components`
- [x] Include relevant code or preferably a [code sandbox](https://codesandbox.io/embed/react-data-table-sandbox-ccyuu
)

## Describe the bug
Selector on nested data doesn't seem to work.

## Expected behavior

When passing in data that has two layers of objects, I expected to be able to use a selector like:
```
const columns = [
    {
        name: 'Bar',
        selector: row => row.foo.bar
    }
]
```

In this case I have a data object like such 
```
data = {
   foo: {
      bar: "data to display"
   }
}
```

However when I run it, I get the message that `row.foo is undefined`, even though I've examined `data` and know that it is NOT undefined-- it's an object.

## Versions (please complete the following information)
 - React: 17.0.2
 - Styled Components: 5.3.3
 - OS: Windows
 - Browser: Firefox

## Additional context

I know this might be an unintended usage of selectors, but I'd find it very useful in order to take a large object from a data source and select what data I want to be in each column. I could potentially open a PR to fix this if supporting nested selectors is OK. Thank you for creating this useful component!

**Comment by NathanC** [comment]
Augh, I was incorrect. The data I was using **was** missing the second-level object. I was pulling from a data source which had some bad data, and the second level was missing in some rows but not others (and not the rows I examined). I normalized the data and it works fine now. Sorry to waste your time. 

tl;dr; Looks like nested selectors work fine (though a small improvement could be not displaying the column data if there's an undefined selector, instead of failing)
